components/form/UploadImageInput.tsx

```diff
@@ -54,33 +56,38 @@ const UploadImageInput: React.FC<UploadImageProps> = ({
 
       if (file) {
         try {
-          const { data: listData, error: listError } = await supabase.storage
-            .from('images')
-            .list(userId, { recursive: true });
+          if (imageType === 'profile') {
+            const { data: listData, error: listError } = await supabase.storage
+              .from('images')
+              .list(userId, { recursive: true });
 
-          if (listError) {
-            console.error('Error fetching old images:', listError);
-            return;
-          }
+            if (listError) {
+              console.error('Error fetching old images:', listError);
+              return;
+            }
 
-          if (!listData) {
-            throw new Error('Failed to fetch images');
-          }
+            if (!listData) {
+              throw new Error('Failed to fetch images');
+            }
 
-          const oldImages = listData.map((image) => image.name);
+            const profileImages = listData
+              .filter((image) => image.name.startsWith('profile_'))
+              .map((image) => userId + '/' + image.name);
 
-          if (oldImages.length > 0) {
-            const { error: deleteError } = await supabase.storage
-              .from('images')
-              .remove(oldImages.map((image) => userId + '/' + image));
+            if (profileImages.length > 0) {
+              const { error: deleteError } = await supabase.storage
+                .from('images')
+                .remove(profileImages);
 
-            if (deleteError) {
-              console.error('Error deleting old images:', deleteError);
-              return;
+              if (deleteError) {
+                console.error('Error deleting old images:', deleteError);
+                return;
+              }
             }
           }
 
-          const imageName = uuidv4();
+          const prefix = imageType === 'profile' ? 'profile_' : 'item_';
```

Comment:
we could probably just interpolate the actual imageType since it's being controlled by the type declaration and it would make it a nicer read for the future

```typescript
const prefix = `${imageType}_`
```

Reply:
nice suggestion - thank you!

Reply:
... but after working on this for more than two hours, cannot make it work 😫. I added the following lines of code :
```javascript
console.log('imageType', imageType);
const prefix = `${imageType}_`;
```
But I keep getting "imageType undefined" for 'item'. I think there might be an issue with how the imageType prop is being handled from AddNewItemForm to UploadImageInput. I haven't been able to figure out what the problem is yet, but it's probably something very simple  😄 

I'll leave this as it is for now.

Reply:
This is odd and makes me really curious, but I'm running everything fine and getting the expected urls for image uploads 🤔 

Since we're adding tests as well to check on this feature I feel pretty confident in what we have.